lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -167,17 +167,17 @@ SerializedModuleLoaderBase::findModule(AccessPathElem moduleID,
   ModuleFilenamePair fileNames(moduleName);
 
   StringRef archName = Ctx.LangOpts.Target.getArchName();
-  ModuleFilenamePair archFileNames(archName);
+
+  SmallVector<ModuleFilenamePair, 4> targetFileNamePairs;
+  targetFileNamePairs.push_back(archName);
 
   // FIXME: We used to use "major architecture" names for these files---the
   // names checked in "#if arch(...)". Fall back to that name in the one case
   // where it's different from what Swift 4.2 supported: 32-bit ARM platforms.
   // We should be able to drop this once there's an Xcode that supports the
   // new names.
-  StringRef alternateArchName;
   if (Ctx.LangOpts.Target.getArch() == llvm::Triple::ArchType::arm)
-    alternateArchName = "arm";
-  ModuleFilenamePair alternateArchFileNames(alternateArchName);
+    targetFileNamePairs.push_back(StringRef("arm"));
```

Comment:
The next commit will change `StringRef("arm")` to `ModuleFilenamePair("arm")` to improve clarity. (Forgot to amend that in.)